### PR TITLE
Update Dockerfile and fix build problem

### PR DESCRIPTION
**Pull request by vowstar**
The old Dockerfile have some problem while building NNPACK, such as missing confu, mismatched ninja version.

Before fix

```bash
...
Cloning into 'NNPACK'...
Traceback (most recent call last):
File "./configure.py", line 4, in <module>
import confu
ImportError: No module named confu
Removing intermediate container a7d5d50c6b88
The command '/bin/sh -c apt-get install ninja-build && pip install ninja-syntax && cd /software && git clone --recursive https://github.com/Maratyszcza/NNPACK.git && cd /software/NNPACK && python ./configure.py && ninja' returned a non-zero code: 1
```

After fix

```bash
...
[152/156] LINK bin/convolution-inference+relu-smoketest
[153/156] CXX test/softmax-output/smoke.cc
[154/156] LINK bin/softmax-output-smoketest
[155/156] CXX test/max-pooling-output/smoke.cc
[156/156] LINK bin/max-pooling-output-smoketest
---> 60abb17a95c2
Removing intermediate container 0b970dd3258e
# build success
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04
+FROM ubuntu:17.04
```
Comment (Randl) [inline Dockerfile]: Should we stay on LTS?

Comment (vowstar) [inline Dockerfile]: The ninja version is too old in TLS, NNPACK can't compile. Docker CE stable version only support 1 year, so it's meaningless for using ubuntu LTS.

Comment (edgarriba) [inline Dockerfile]: ping @bhack

Comment (bhack) [inline Dockerfile]: Can we wait 13th April for merge this?

Comment (vowstar) [inline Dockerfile]: ping @bhack

**Pull request merged**